providers/implementations/kem/rsa_kem.c

```diff
@@ -329,13 +347,24 @@ static int rsasve_recover(PROV_RSA_CTX *prsactx,
         return 1;
     }
 
-    /* Step (2): check the input ciphertext 'inlen' matches the nlen */
+    /*
+     * Step (2): check the input ciphertext 'inlen' matches the nlen
+     * and that outlen is at least nlen bytes
+     */
     if (inlen != nlen) {
         ERR_raise(ERR_LIB_PROV, PROV_R_BAD_LENGTH);
         return 0;
     }
+    if (outlen == NULL || *outlen < nlen) {
+        ERR_raise(ERR_LIB_PROV, PROV_R_INVALID_OUTPUT_LENGTH);
+        return 0;
+    }
+
     /* Step (3): out = RSADP((n,d), in) */
-    return (RSA_private_decrypt(inlen, in, out, prsactx->rsa, RSA_NO_PADDING) > 0);
+    ret = RSA_private_decrypt(inlen, in, out, prsactx->rsa, RSA_NO_PADDING);
+    if (ret > 0)
+        *outlen = ret;
+    return ret > 0;
```

Comment:
same here, ret == 0 means no error.

Reply:
same as above.  There is likely an argument to be made here, but since the code before the change treated zero returns from RSA_private_decrypt as an error, I would like to keep that in place until after the release unless there is a specific outstanding bug about it.